openssl-sys/src/lib.rs

```diff
@@ -374,30 +154,21 @@ pub const SSL_VERIFY_NONE: c_int = 0;
 pub const SSL_VERIFY_PEER: c_int = 1;
 pub const SSL_VERIFY_FAIL_IF_NO_PEER_CERT: c_int = 2;
 
-pub const SSL_OP_MICROSOFT_SESS_ID_BUG: c_long =                    0x00000001;
-pub const SSL_OP_NETSCAPE_CHALLENGE_BUG: c_long =                   0x00000002;
-pub const SSL_OP_NETSCAPE_REUSE_CIPHER_CHANGE_BUG: c_long =         0x00000008;
-pub const SSL_OP_TLSEXT_PADDING: c_long =                           0x00000010;
-pub const SSL_OP_MICROSOFT_BIG_SSLV3_BUFFER: c_long =               0x00000020;
-pub const SSL_OP_SSLEAY_080_CLIENT_DH_BUG: c_long =                 0x00000080;
-pub const SSL_OP_TLS_D5_BUG: c_long =                               0x00000100;
-pub const SSL_OP_TLS_BLOCK_PADDING_BUG: c_long =                    0x00000200;
-pub const SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS: c_long =              0x00000800;
-pub const SSL_OP_ALL: c_long =                                      0x80000BFF;
-pub const SSL_OP_NO_QUERY_MTU: c_long =                             0x00001000;
-pub const SSL_OP_COOKIE_EXCHANGE: c_long =                          0x00002000;
-pub const SSL_OP_NO_TICKET: c_long =                                0x00004000;
-pub const SSL_OP_CISCO_ANYCONNECT: c_long =                         0x00008000;
-pub const SSL_OP_NO_SESSION_RESUMPTION_ON_RENEGOTIATION: c_long =   0x00010000;
-pub const SSL_OP_NO_COMPRESSION: c_long =                           0x00020000;
-pub const SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION: c_long =        0x00040000;
-pub const SSL_OP_SINGLE_ECDH_USE: c_long =                          0x00080000;
-pub const SSL_OP_SINGLE_DH_USE: c_long =                            0x00100000;
-pub const SSL_OP_CIPHER_SERVER_PREFERENCE: c_long =                 0x00400000;
-pub const SSL_OP_TLS_ROLLBACK_BUG: c_long =                         0x00800000;
-pub const SSL_OP_NO_SSLv2: c_long =                                 0x01000000;
-pub const SSL_OP_NO_SSLv3: c_long =                                 0x02000000;
-pub const SSL_OP_NO_TLSv1: c_long =                                 0x04000000;
+#[cfg(not(ossl101))]
```

Comment:
I was thinking that for a particular OpenSSL version this gives you what *can* be the API of that version, although it'd ideally be trimmed to the precise API of what's exactly on the system.

In that sense I'm definitely in favor of conditionally compiling away all the OpenSSL 1.1.0 APIs when compiling for OpenSSL 1.0.1, for example, to weed out errors sooner.

(also this is required to pass systest)